Start quiz page

Dumitrescu-Dragos opened the pull request:
This is only the first part of the feature

Dumitrescu-Dragos committed:
Add HttpService and StorageService

andiabrudan committed:
AddedPassword Encryption with SHA256

andiabrudan committed:
Updated Encryptor. It can now be used as an extension.

andiabrudan committed:
Now session token only holds user.Id
Added extension for getting user.Id from claims

louisavasies committed:
Added Red Border for Error

andiabrudan committed:
Woops forgot the extension method

Dumitrescu-Dragos committed:
Refactor components:
- SubmitButton.tsx and SubmitButton.css
- FormInput.tsx
- App.tsx (Add route for testing)

Dumitrescu-Dragos committed:
Refactor components:
- SubmitButton.tsx and SubmitButton.css
- FormInput.tsx
- App.tsx (Add route for testing)

Dumitrescu-Dragos committed:
Add new assets to media

Dumitrescu-Dragos committed:
Add QuizComponent and StartQuizComponent

Dumitrescu-Dragos committed:
Add QuizComponent and StartQuizComponent

Dumitrescu-Dragos committed:
Refactor App.tsx and add not-found.png

Dumitrescu-Dragos committed:
Refactor App.tsx and add not-found.png

Dumitrescu-Dragos committed:
Solve conflicts

Dumitrescu-Dragos committed:
Solve conflicts

Dumitrescu-Dragos committed:
Solve conflicts

Dumitrescu-Dragos committed:
Solve conflicts

Dumitrescu-Dragos committed:
Refactor QuizComponent and StartQuizComponent

Dumitrescu-Dragos committed:
Add NotFoundComponent

Dumitrescu-Dragos committed:
Add NotFoundComponent

Dumitrescu-Dragos committed:
Fix App.tsx

Dumitrescu-Dragos committed:
Fix App.tsx

Dumitrescu-Dragos committed:
Add HttpService and Refactr Login and Signup

louisavasies committed:
Added Code Review Changes

louisavasies committed:
Start Game Page

andrianwritek committed:
Merge pull request #3 from andrianwritek/services

Services

Commit:
Image upload testing

louisavasies committed:
Final change

andrianwritek committed:
Merge pull request #1 from andrianwritek/INT2018-34_Red_Border_For_Errors

Added Red Border for Error

andiabrudan committed:
Changed token generator function to an extension method.

Commit:
Empty quiz test route

andiabrudan committed:
Email Controller: added support for reseting password

Dumitrescu-Dragos committed:
Update StartQuizComponent:
- add send quiz request
- add preview

louisavasies committed:
Added kebab-case for all CSS; constant message in StartGame page.

andiabrudan committed:
Fixed issues found in pull request

Dumitrescu-Dragos committed:
Refactor UserExtensions.cs

Dumitrescu-Dragos committed:
Refactor UserExtensions.cs

Dumitrescu-Dragos committed:
Refactor NotFoundComponent, FormInput and App Components:
- Move footer to App.tsx
- Refactor NotFoundComponent (capitalize name and delete foooter)
- Refactor FormInput (capitalize name)
- Remove footer from existing components (Login, Signup)

Dumitrescu-Dragos committed:
Refactor NotFoundComponent, FormInput and App Components:
- Move footer to App.tsx
- Refactor NotFoundComponent (capitalize name and delete foooter)
- Refactor FormInput (capitalize name)
- Remove footer from existing components (Login, Signup)

andiabrudan committed:
Fixed issues found in pull request #2

andiabrudan committed:
Fixed issues found in pull request #3

Commit:
Create/update the quiz based on the image url and a title

andiabrudan committed:
Revert "Changed token generator function to an extension method."

This reverts commit 3a68e2e02a95abc42224cc0e919b08fc282fe435.

Dumitrescu-Dragos committed:
Slve conflicts

Commit:
Minor Quiz Create/Update refactor

Commit:
Minor naming changed

andiabrudan committed:
Fixed isssues found in pull request #4
pwease merge my pwull wequest T.T

Dumitrescu-Dragos committed:
Use Interfaces instead of classes in entities

Dumitrescu-Dragos committed:
Add validations in StartQuizComponent

andiabrudan committed:
Merge branch 'develop' into INT2018-35_Password_Encryption

Commit:
Modified Quizzes Controller and Quiz repo to also verifiy the user id

andiabrudan committed:
Fixed isssues found in pull request #5

Dumitrescu-Dragos committed:
Merge branch 'notFoundPage' of https://github.com/AvaSpa/VikinQuiz into notFoundPage

# Conflicts:
#	VikingQuiz.UI/src/Components/FormComponent/FormInput/FormInput.tsx

Dumitrescu-Dragos committed:
Refactor QuizComponent and StartQuizComponent:
- add httpService
- add validation

Dumitrescu-Dragos committed:
Add QuestionsComponent

vasi-axinte committed:
Merge pull request #4 from andrianwritek/notFoundPage

Not found page

andiabrudan committed:
Merge branch 'develop' into INT2018-35_Password_Encryption

Commit:
Fixed URL return value | Added the ability to delete the old photo when upadting a quiz

Dumitrescu-Dragos committed:
Update StartQuizComponent.tsx, QuizComponent.tsx, FormInput.tsx

louisavasies committed:
Added Play Button, resolved PlayerController for GET; Almost Finished StartGame Page

Commit:
Modified UPDATE quiz route to also check the user ID

Commit:
- added Migration for project
- added new field "Code" in Game table
- reverted "hack" for migrations

vasi-axinte committed:
Merge pull request #2 from andrianwritek/INT2018-35_Password_Encryption

Int2018 35 password encryption

Commit:
End of day commit (name changes)

Dumitrescu-Dragos committed:
Delete submit button and make components read-only

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
+    
+        private IActionResult FileValidityChecker(NewQuizViewModel quizBodyData)
+        {
+            IActionResult statusCodeResult = Ok();
```
This code is repeating. There is no need for that

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
+            IActionResult statusCodeResult = Ok();
+            statusCodeResult = Ok();
+            if (quizBodyData.Files.Count == 0) {
+                statusCodeResult =  BadRequest("An image file need to be provided."); }
```
I think that all these checks can be extracted in separate private methods that would return an IActionResult instead of writing so much code in a controller method. Endpoints in a controller shall have as little code as possible.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
         {
-            var quizzes = quizRepo.GetAll();
-            return Ok(quizzes.Select(quiz => this.entityToVmMapper.Map(quiz)));
+            // User.Claims.GetUserId(); -- use this to get the ID of the user based on the token recieved
```
Please remove commented code

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
+            AzureBlobService blobService = new AzureBlobService();
+            await blobService.InitializeBlob();
+
+            var quizzes = quizRepo.GetAll(3); // pass that user ID in
```
Why do we always pass the number 3 to the GetAllMethod?

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
+                blobService = new AzureBlobService();
+                await blobService.InitializeBlob();
+                fileUrl = await blobService.UploadPhoto(quizBodyData.Files[0]);
+            } catch (Exception) {
```
Usually it is a bad practice to catch Exception without take some information from it on one side. On the other side it is also a bad practice to just catch "Exception" instead of something specific.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
-        public IActionResult Add([FromBody]QuizViewModel quiz)
+        [RequestSizeLimit(5_000_000)]
+        //[Authorize] add authorize when avaiable on the front-end
+        public async Task<IActionResult> Post(NewQuizViewModel quizBodyData)
```
This method is too long. Please, refactor it into smaller methods that are called here.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepo quizRepo, IEntityMapper<QuizViewModel, Quiz> v
+        [HttpPost("{id}")]
+        [RequestSizeLimit(5_000_000)]
+        //[Authorize] add authorize when avaiable on the front-end
+        public async Task<IActionResult> Put(NewQuizViewModel quizBodyData, int id)
```
This method is too long also.

vasi-axinte commented on VikingQuiz.Api/Controllers/ValuesController.cs:
```diff
@@ -1,35 +1,53 @@
 namespace VikingQuiz.Api.Controllers
 {
     [Route("api/[controller]")]
     public class ValuesController : Controller
```
Is this controller really used for something?

vasi-axinte commented on VikingQuiz.Api/Repositories/QuizRepo.cs:
```diff
@@ -37,18 +37,30 @@ public Quiz UpdateQuiz(Quiz quiz)
+
             var games = ctx.Game.Where(x => x.QuizId == id).ToList();
             ctx.Game.RemoveRange(games);
             var quizquestions = ctx.QuizQuestion.Where(x => x.QuizId == id).ToList();
```
Please use camelCase for this name since it contains 2 words

vasi-axinte commented on VikingQuiz.Api/Repositories/QuizRepo.cs:
```diff
@@ -37,18 +37,30 @@ public Quiz UpdateQuiz(Quiz quiz)
+        public Quiz DeleteQuiz (int id, int userId)
         {
+            // check if the player has acces to that quiz in order to delete it
             Quiz quiz = new Quiz
```
Why do we need to create this quiz?

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+namespace VikingQuiz.Api.Utilities
+{
+    public class AzureBlobService
+    { // should be changed to a SINGLETON if posibile
```
Please remove this comment and put a task in Jira instead. You can call it "Investigate if AzureBlobService can be made a singleton" or something similar

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+        private CloudBlobClient blobClient;
+        private ServiceProperties serviceProperties;
+        private CloudBlobContainer userContainer;
+        public Uri urlPath;
```
Public fields should go first. Also, use properties instead of public fields. 
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+
+        public async Task<bool> InitializeBlob()
+        {
+            userContainer = blobClient.GetContainerReference("users");
```
The string "users" should be extracted in some constant or enum if there are many.

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+        public async Task<bool> InitializeBlob()
+        {
+            userContainer = blobClient.GetContainerReference("users");
+            if (!await userContainer.CreateIfNotExistsAsync())
```
Why is this method called twice?

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+            }
+
+            string contentType = file.ContentType;
+            var extension = contentType == "image/png" ? ".png" :
```
Instead of this ugly sintax, you can use either a Dictionary for contentType or some other method which would return you the right type.

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+                            contentType == "image/jpeg" ? ".jpeg" :
+                            contentType == "image/gif" ? ".gif" : null;
+
+            string imgName = Guid.NewGuid().ToString() + (Int32)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds + extension;
```
Please don`t use abreviations in naming variables.

vasi-axinte commented on VikingQuiz.Api/Utilities/AzureBlobService.cs:
```diff
@@ -0,0 +1,109 @@
+        public AzureBlobService()
+        {
+            // use TryParse to check connection, return exception if it can't connect
+            //account = CloudStorageAccount.DevelopmentStorageAccount;
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/LoginFormComponent/LoginFormComponent.tsx:
```diff
@@ -11,13 +11,13 @@ class LoginFormComponent extends FormComponent {
+        {this.state.inputs.map((input: IInputData) => this.renderInput(input))}
         <div className="forget-remember col-xs-12">
-          <a href="#" className="forget-pass col-xs-6">FORGOT PASSWORD</a>
+          <a href="#" className="forget-pass col-xs-6">forgot password</a>
```
Please extract all these strings in some constants. Using constants, it will make your life easier when you will need to change the text or add Localization (translated text based on browser language)

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsListComponent/QuestionsListComponent.tsx:
```diff
@@ -0,0 +1,20 @@
+    public render(){
+        return (
+            <div className='questions-list'>
+                <h3>Questions List</h3>
```
Please remove these kind of hardcoded divs and only put relevant data in files.

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.css:
```diff
@@ -0,0 +1,111 @@
+    font-size: 27px;
+}
+
+/* title */
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.css:
```diff
@@ -0,0 +1,111 @@
+    margin-left: 10px;
+}
+
+/* errors */
```
also here

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,273 @@
+                const imageUrl: string = response.data.pictureUrl;
+                const titleValue: string = response.data.title;
+
+                console.log(imageUrl, titleValue);
```
Please don`t leave console logs into your code.

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,273 @@
+            const getQuizUrl: string = 'http://localhost:60151/api/quizzes/' + this.props.quizId; 
+            this.httpService.getWithToken(getQuizUrl)
+            .then((response: any) => {
+                // this part gets the photo at the specified url
```
We know, please remove the comment :)

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,273 @@
+
+    public componentWillMount(){
+        if(this.props.editMode){
+            const getQuizUrl: string = 'http://localhost:60151/api/quizzes/' + this.props.quizId; 
```
The url should be stored in a constant.

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,273 @@
+        this.fileInputRef.current.click();
+    }
+
+    public saveQuizHandler = () => {
```
This method is quite long. Can we maybe refactor it a little bit?

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,273 @@
+
+        let url: string = '';
+        if(this.props.editMode){
+            url = 'http://localhost:60151/api/quizzes/' + this.props.quizId;
```
If you would have stored the url in a constant, it would have been way easier to change it in all places at once.

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,273 @@
+        return true;
+    }
+
+    private allValid = (): boolean => {
```
This method name doesn`t suggest me too many things. Please find a more meaningful name

andiabrudan committed:
Added SendEmailComponent and ResetPasswordComponent

andiabrudan committed:
Changed from using HttpService to using axios

andiabrudan committed:
Added validation for ResetPasswordComponent

louisavasies committed:
Made page responsive; Added Start and Cancel Game Buttons

louisavasies committed:
Changed algorithm for showing players; Deleted comments

Dumitrescu-Dragos committed:
Refactor StartQuiz.tsx

Dumitrescu-Dragos committed:
Refactor Mappers and QuestionController:
- Mappers: AnswerMapper and QuestionMapper

Dumitrescu-Dragos committed:
Add QuestionController backend functionality

andiabrudan committed:
Fixed issues found in pull request #1

andiabrudan committed:
Added MyProfileComponent

vasi-axinte committed:
Merge pull request #8 from andrianwritek/INT2018-51_My_Profile_Component

Added MyProfileComponent

Dumitrescu-Dragos committed:
Add Snackbar component

Dumitrescu-Dragos committed:
Update StartQuizComponent to use SnackbarComponent

Dumitrescu-Dragos committed:
Update SnackbarComponent

andiabrudan committed:
Helped Dragos with server error messages

Dumitrescu-Dragos committed:
Update storageService and QuizComponent for Protected Routes

andiabrudan committed:
Fixed issues found in pull request #2

Dumitrescu-Dragos committed:
Add PrivateRoute Component (commented)

AndronicMonica committed:
MyQuizzes page - frontend and backend

Dumitrescu-Dragos committed:
Fixed rebase conflicts

Dumitrescu-Dragos committed:
Fix Snackbar Component

Dumitrescu-Dragos committed:
Fix Update Question method

Dumitrescu-Dragos committed:
Solve bug: duplicate questions after update

Commit:
Commutative front-end changes for the edit/add answers

vasi-axinte commented on VikingQuiz.Api/Controllers/AnswerController.cs:
```diff
@@ -50,8 +43,7 @@ public IActionResult CreateAnswer([FromBody]AnswerViewModel answer)
         {
             Answer ans = new Answer
```
Please try to avoid abbreviations like "ans"

vasi-axinte commented on VikingQuiz.Api/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,149 @@
+    [Route("api/[controller]")]
+    public class QuestionsController : Controller
+    {
+        private readonly QuestionRepository questionRepo;
```
These should be renamed to questionRepository and answerRepository

vasi-axinte commented on VikingQuiz.Api/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,149 @@
+        {
+            var questions = questionRepo.GetAllByQuizId(quizId);
+            var result = questions
+                        .Select(question => {
```
This select does not look very nice. Please avoid using return statements inside LINQ.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,149 @@
+
+        [HttpPost]
+        public IActionResult CreateQuestion([FromBody]QuestionViewModel questionVm, [FromQuery(Name = "quiz")] int quizId)
+        {
```
This method is too long. You should separate the concerns within into several private methods.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,149 @@
+                return BadRequest("The text is already assigned to another question");
+            }
+
+            List<AnswerViewModel> realAnswers = new List<AnswerViewModel>();
```
New variables should be declare at the top of a method

vasi-axinte commented on VikingQuiz.Api/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,149 @@
+                    return BadRequest("You can't have two duplicate answers");
+                }
+
+                AnswerViewModel newAnswerVM = this.answerToVmMapper.Map(newAnswer);
```
This should be named newAnswerVm. Take care at naming conventions.

Dumitrescu-Dragos committed:
Refactor Backend QuizController and QuestionController to use Token
Note: fix validation on model in Entity Framework

Dumitrescu-Dragos committed:
Refactor StartQuizComponent

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepository quizRepository, IEntityMapper<QuizViewMo
+            AzureBlobService blobService = new AzureBlobService();
+            await blobService.InitializeBlob();
+
+            var quizzes = quizRepository.GetAll(3); // pass that user ID in
```
Please remove the comment and change the parameter passed to that function. We should avoid sending magic numbers in method calls.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepository quizRepository, IEntityMapper<QuizViewMo
         }
 
+
+        // Get a specific quiz id only if it belongs to that user
```
Please remove this kind of comments.

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepository quizRepository, IEntityMapper<QuizViewMo
+                return imageHttpResponse;
+            }
+
+            string fileUrl;
```
variables should be declared at the top of method.

Dumitrescu-Dragos committed:
Fix Snackbar Show

Commit:
Fixed edit questions page to be hidden by default

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepository quizRepository, IEntityMapper<QuizViewMo
             {
+                Title = quizBodyData.Title,
+                PictureUrl = fileUrl,
+                UserId = 3
```
Why is this magic number assigned by default as the UserId?

vasi-axinte commented on VikingQuiz.Api/Controllers/QuizzesController.cs:
```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepository quizRepository, IEntityMapper<QuizViewMo
-        [HttpPut]
-        public IActionResult Update([FromBody]QuizViewModel quiz)
+
+        // update a specific quiz id
```
Please remove this kind of comments.

vasi-axinte commented on VikingQuiz.Api/Repositories/AnswerRepository.cs:
```diff
@@ -17,31 +17,37 @@ public AnswerRepository(VikinQuizContext context)
-            existingAnswer.QuestionId = ans.QuestionId;
+            existingAnswer.QuestionId = questionId;
             context.SaveChanges();
             return ans;
```
This should be named "answer"

vasi-axinte commented on VikingQuiz.Api/Repositories/QuizRepository.cs:
```diff
@@ -23,7 +23,7 @@ public Quiz CreateQuiz(Quiz quiz)
         public Quiz UpdateQuiz(Quiz quiz)
         {
-            Quiz foundQuiz = context.Quiz.Where(x => x.Id == quiz.Id).FirstOrDefault();
+            Quiz foundQuiz = context.Quiz.Where(x => x.Id == quiz.Id && x.UserId == quiz.UserId).FirstOrDefault();
```
Can we maybe use a more meaningful letter than "x" inside the Where clause?

vasi-axinte commented on VikingQuiz.Api/Repositories/QuizRepository.cs:
```diff
@@ -37,18 +37,30 @@ public Quiz UpdateQuiz(Quiz quiz)
-        public void DeleteQuiz(int id)
+        public Quiz DeleteQuiz (int id, int userId)
         {
+            // check if the player has acces to that quiz in order to delete it
```
Please avoid this type of comments.

Dumitrescu-Dragos committed:
Add postSuccess and postError

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+}
+class QuestionsComponent extends React.Component<IProps, IState> {
+
+   //#region fields
```
please remove the region.

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+   public readonly endpointUrl = "http://localhost:60151/api/questions";
+   public paramsObject = {
+      params: {
+         quiz: this.props.quizId // quizId 
```
Why do we need that comment?

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+   }
+   public state = {
+      isEditMode: false,
+      activeQuizId: this.props.quizId, // this.props.quizId
```
Why do we need that comment?

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+   public getReferenceToRefreshQuestionList = () => {
+      return this.refreshQuestionList;
+   }
+
```
Why so many blank lines?

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+   }
+
+   public deleteButtonClickHandler = (question : any) => {
+      // this.deleteQuestionRequest(this.state.activeQuizId);
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+
+   }
+
+   // reset the state to its default values
```
Please get rid of this type of comments.

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsComponent.tsx:
```diff
@@ -0,0 +1,356 @@
+         const reqUrl = this.endpointUrl + "?quiz=" + this.props.quizId;
+         const request = this.httpService.post(reqUrl, newQuestionData);
+         if (request) {
+            console.log("-------------CREATE NEW QUIZ REQUEST SENT---------------");
```
Please avoid leaving console logs through your code

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsListComponent/QuestionsListComponent.tsx:
```diff
@@ -0,0 +1,183 @@
+   public questionsToJsx = (questionsData: any[]) => {
+      let jsxElements: any;
+
+      // questionsData = questionsData.filter( (question) => question.id !== correctAnswerId);
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/QuestionsComponent/QuestionsListComponent/QuestionsListComponent.tsx:
```diff
@@ -0,0 +1,183 @@
+
+
+   // public dummyQuestionsData = [
+   //    {
```
Please remove all these commented lines

vasi-axinte commented on VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx:
```diff
@@ -0,0 +1,315 @@
+    }
+
+    private isImageValid = (fileObject: any): boolean => {
+        const acceptedFormats: any = ['png','jpeg', 'gif'].map((type: string) => 'image/' + type);
```
This method can be split in several smaller methods

vasi-axinte commented on VikingQuiz.UI/src/Components/SignUpPage/SignUpPage.tsx:
```diff
@@ -1,11 +1,11 @@
+import IUserDto from '../../entities/IUserDto';
 import { Redirect } from 'react-router-dom';
 import {signUpRules} from '../../entities/Validation/rules';
 // import register from '../../registerServiceWorker';
```
Please remove unused imports

vasi-axinte commented on VikingQuiz.UI/src/Components/SignUpPage/SignUpPage.tsx:
```diff
@@ -92,10 +91,10 @@ class SignUpPage extends React.Component<{}, any> {
+                                {id: 'user-password', type: 'password', label: 'Password', errorMessage: '', name: 'Password', value: ''},
+                                {id: 'user-confpass', type: 'password', label: 'Confirm Password', errorMessage: '', name: 'ConfPassword', value: ''}
                            ]}
                            url="http://localhost:60151/api/users"
```
Why is this still not extracted in a constant?

Commit:
Added Confirmation box to the delete button

Dumitrescu-Dragos committed:
- Fix LoginPage and SignupPage design
- Fix login with Google and Facebook

Commit:
Minor changes (to be up to date)

louisavasies committed:
Finished pages for creating a player for a game in progress and connectig the player to it;
Added PlayerGameViewModel and PlayerGameController+Repo + Mappers;
Added DTO for player+code;
Finished PlayerController for having codes;
Changed GameController with updated version;

Dumitrescu-Dragos committed:
Refactor AnswerController and QuestionsController

Dumitrescu-Dragos committed:
Refactor QuestionsListComponent.tsx and StartQuizComponent.tsx

andiabrudan committed:
Added functionality for "Remember Me" checkbox

Dumitrescu-Dragos committed:
Refactor frontend:
- delete comments
- delete console.logs

Commit:
Code review changes

AndronicMonica committed:
MyQuizzes Page - modified based on code review

Commit:
Minor answers input styling changes

AndronicMonica committed:
MyQuizzesPage - quiz and user controller with authorize

Commit:
Minor fixes for questions page

vasi-axinte committed:
Merge pull request #9 from andrianwritek/INT2018-MyQuizes

MyQuizzes page - frontend and backend

louisavasies committed:
Applied changes for code review.

louisavasies committed:
Fixed rebase errors.

andiabrudan committed:
Fixed issues found in pull request

vasi-axinte committed:
Merge pull request #10 from andrianwritek/INT2018-52_Remember_Me

Added functionality for "Remember Me" checkbox

andiabrudan committed:
Fixed issues found in pull request #3

vasi-axinte committed:
Merge pull request #7 from andrianwritek/INT2018-23-48_Password_Reset

Int2018 23 48 password reset

Dumitrescu-Dragos committed:
Rebase startQuiz and develop

Dumitrescu-Dragos committed:
Refactor Monica's Components

Commit:
Added logout button to the quiz component page

louisavasies committed:
Changed some variable names in Repositories;
Corrected Front-End on StartGame&Connect pages;

Dumitrescu-Dragos committed:
Fix SnackbarComponent and Login with Facebook and Google

louisavasies committed:
Renamed variable playergameRepo

louisavasies committed:
Fixed validation rules

vasi-axinte committed:
Merge pull request #5 from andrianwritek/INT2018-10-Create-Quiz-For-Friends

Create Quiz For Friends

AndronicMonica committed:
Redirect and manageQuiz - fixed

Commit:
Redesign My profile

Dumitrescu-Dragos committed:
Refactor Login and Singup page

Dumitrescu-Dragos committed:
Fix design of the app
